lnrpc/autopilotrpc/autopilot_server.go

```diff
@@ -0,0 +1,204 @@
+// +build autopilotrpc
+
+package autopilotrpc
+
+import (
+	"context"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"sync/atomic"
+
+	"github.com/lightningnetwork/lnd/autopilot"
+	"github.com/lightningnetwork/lnd/lnrpc"
+	"google.golang.org/grpc"
+	"gopkg.in/macaroon-bakery.v2/bakery"
+)
+
+const (
+	// subServerName is the name of the sub rpc server. We'll use this name
+	// to register ourselves, and we also require that the main
+	// SubServerConfigDispatcher instance recognize as the name of our
+	subServerName = "AutopilotRPC"
+)
+
+var (
+	// macaroonOps are the set of capabilities that our minted macaroon (if
+	// it doesn't already exist) will have.
+	macaroonOps = []bakery.Op{
+		{
+			Entity: "autopilot",
```

Comment:
So one question is if we want to require an _entirely_ new macaroon for interaction with the pilot sub-sever, or if we want to use the existing entity mapping. FWIW, the latter route would allow the `admin.macaroon` to use the service as is without having to generate an additional set of macaroons. However, if we make these a series of "sub-commands" for `lncli`, then these sub-commands can themselves know where to fetch the proper macaroon.

Reply:
I'm fine with reusing existing macaroons, as you don't really gain more access than what you already have with the regular macaroon.

However, it would be nice if subservers had a standardized macaroon setup.

Reply:
Yeah re macaroon set up, once the primary sub-servers are in, following your suggestion, we can modify them to simply present which macaroons they need, and then the (to be fleshed out) bakery would handle baking them.

Reply:
Had some thoughts about this earlier w.r.t a "catch all" capability in the admin macaroon. This is a follow up though, and would require one last (hopefully) change to the admin macaroon to basically give it a "God" capability. All other method calls would by pass their regular checks if a macaron with this capability was presented.

Reply:
Ended up removing custom autopilot macaroons completely for now, and instead require the `info/read` permission for getting status, and `offchain/write + onchain/write` for modifying the status.